docs/core/howto/ssl.rst

```diff
@@ -204,10 +199,40 @@ For example,
     options = CertificateOptions(..., dhParameters=dhParams)
 
 Another part of the TLS protocol which ``CertificateOptions`` can control is the version of the TLS or SSL protocol used.
-This is often called the context's "method".
+There are two ways to configure it, either by configuring a list of protocols, or by the legacy way of setting the internal context's "method".
+By default, Twisted will configure it to use TLSv1.0 or later and disable the insecure SSLv3 protocol, but this manual control over protocols can be helpful if you need to support legacy SSLv3 systems, or you wish to restrict it down to just the strongest of the TLS versions.
+
+You can give ``CertificateOptions`` a range of TLS versions you wish it to negotiate with the ``tlsProtocols`` argument in the initializer:
+
+.. code-block:: python
+
+    from twisted.internet.ssl import (CertificateOptions, TLSVersion,
+                                      TLSVersionRange)
+    options = CertificateOptions(
+        ...,
+        tlsProtocols=TLSVersionRange(TLSVersion.TLSv1_0,
+                                     TLSVersion.TLSv1_2))
+
```

Comment:
Enumerating all the use-cases like that suggests a totally different API, given how its behavior should change as code ages.

For example, let's say you're using the API for reason 2; you want to be *more* secure by the default.  If we have an API that expresses that specifically, and you were to do `CertificateOptions(increaseMinimumVersionTo=TLS_v1_3)`, in a few years when TLS 1.3 has been phased out in favor of something new and we've decided to EOL it in Twisted, that option would be a no-op.  However, if the API were a simple enumeration, and you did simply `CertificateOptions(tlsProtocols=set([TLSVersion.TLSv1_1, TLSVersion.TLSv1_2, TLSVersion.TLSv1_3]))` then when TLS 1.5 comes out, if you haven't been maintaining this software, you end up with a _less_ secure configuration, which is the opposite of what you wanted.